### PR TITLE
[autoscaler] Update azure pip packages in the cluster yaml

### LINKED ISSUE #13599: [autoscaler] Azure SDK version needs to be pinned
**TLDR; We need to pin `azure-mgmt-compute==14.0.0`** 

### What is the problem?

While trying to deploy/update a ray cluster in Azure, the following error is generated:

```
$ ray up example-full.yaml 
Cluster: <hidden>

Checking Azure environment settings
2021-01-21 08:27:03,147	INFO config.py:39 -- Using subscription id: <hidden>
2021-01-21 08:27:03,147	INFO config.py:53 -- Creating/Updating Resource Group: <hidden>
Updating cluster configuration and running full setup.
Cluster Ray runtime will be restarted. Confirm [y/N]: y

<1/1> Setting up head node
  Prepared bootstrap config
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/updater.py", line 124, in run
    self.do_update()
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/updater.py", line 273, in do_update
    self.node_id, {TAG_RAY_NODE_STATUS: STATUS_WAITING_FOR_SSH})
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 29, in wrapper
    return f(self, *args, **kwargs)
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 226, in set_node_tags
    self.compute_client.virtual_machines.update(
AttributeError: 'VirtualMachinesOperations' object has no attribute 'update'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/updater.py", line 127, in run
    self.node_id, {TAG_RAY_NODE_STATUS: STATUS_UPDATE_FAILED})
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 29, in wrapper
    return f(self, *args, **kwargs)
  File "/home/giovanni/.virtualenvs/deploy-azure/lib/python3.7/site-packages/ray/autoscaler/_private/azure/node_provider.py", line 226, in set_node_tags
    self.compute_client.virtual_machines.update(
AttributeError: 'VirtualMachinesOperations' object has no attribute 'update'

  Failed to setup head node.
```

*Ray version and other system information (Python version, TensorFlow version, OS):

OS: ubuntu 18.02
Python: 3.7.1
VirtualEnv: [requirements.txt](https://github.com/ray-project/ray/files/5847827/requirements.txt)

Typically,
```
$ pip install --upgrade pip
$ pip install ray azure-cli
```
[example-full.yaml](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/azure/example-full.yaml) is the same provided by ray team (except by the cluster-name/resource_group/subscription_id fields)

### Reproduction (REQUIRED)
```
$ mkvirtualenv deploy-ray
$ pip install --upgrade pip
$ pip install azure-cli ray
$ read -sp "Azure password: " AZ_PASS && echo && az login --allow-no-subscriptions -u <user_email> -p $AZ_PASS
$ ray up example-full.yaml
```

- [X ] I have verified my script runs in a clean environment and reproduces the issue.
- [ X] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by AmeerHajAli**
closes #13599
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by AmeerHajAli** [comment]
cc @GiovanniGatti @gramhagen

**Comment by gramhagen** [comment]
@AmeerHajAli those are all the right places. fwiw the versions have moved on a bit tho, current versions on pypi:
azure-mgmt-compute = 20.0.0
azure-mgmt-network = 18.0.0
azure-cli-core = 2.22.0

**Comment by AmeerHajAli** [comment]
@gramhagen , if you do not mind, can you please check if this works for you? I don't have azure credentials.

**Comment by gramhagen** [comment]
I will try to find some time, but it might not be until friday this week.

**Comment by AmeerHajAli** [comment]
Sounds good. Yeah, I think if we get this merged this week it would be great!

**Comment by wuisawesome** [comment]
The title says to pin to 14.0.0, but the code looks like it's pinning to 18.0.0?

**Comment by AmeerHajAli** [comment]
@wuisawesome my previous reviewer suggesting updating it to the most recent version

**Review by wuisawesome — APPROVED** [review]
This looks good if 18.0 is ok. 

(A little confused because this comment suggests it shouldn't be this simple though? https://github.com/ray-project/ray/issues/13599#issuecomment-767101111)

**Comment by gramhagen** [comment]
> This looks good if 18.0 is ok.
> 
> (A little confused because this comment suggests it shouldn't be this simple though? [#13599 (comment)](https://github.com/ray-project/ray/issues/13599#issuecomment-767101111))

I think #14787 should have addressed this, but I'll test now to see if anything is missing

**Comment by gramhagen** [comment]
hmm, i take it back, there are still several commands in the node_provider which are not covered by #14787, so we should probably keep it at 14.0.0. sorry for the confusion.

I'll continue testing this weekend

File: docker/ray-deps/Dockerfile [hunk 1]
```diff
@@ -14,8 +14,8 @@ RUN $HOME/anaconda3/bin/pip --no-cache-dir install $(basename $WHEEL_PATH)[all]
         "google-api-python-client==1.7.8" \
         "google-oauth" \
         "kubernetes" \
-        "azure-cli-core==2.4.0" \
-        "azure-mgmt-compute==12.0.0" \
+        "azure-cli-core==2.22.0" \
+        "azure-mgmt-compute==20.0.0" \
         "azure-mgmt-msi==1.0.0" \
-        "azure-mgmt-network==10.1.0"; fi) \
+        "azure-mgmt-network==18.0.0"; fi) \
```
Comment (gramhagen) [inline docker/ray-deps/Dockerfile:20]: need to roll back mgmt-compute and mgmt-network for all files to following verisions

```
    "azure-cli-core==2.22.0"
    "azure-mgmt-compute==14.0.0" 
    "azure-mgmt-msi==1.0.0"
    "azure-mgmt-network==10.2.0"
```

**Review by gramhagen — CHANGES_REQUESTED** [review]
need to roll back azure versions, new sdks are still not fully supported

**Comment by AmeerHajAli** [comment]
@gramhagen , can you please rereview this?

**Review by gramhagen — APPROVED** [review]
Looks good

**Comment by AmeerHajAli** [comment]
the failing test is flakey and orthogonal to this change. Merging...

**Pull request merged**